src/states/common/base.rs

```diff
@@ -401,7 +394,7 @@ pub fn from_network_bytes(data: NetworkBytes) -> Result<Message, RoutingError> {
     let result = serialisation::deserialise(&data[..]).map_err(RoutingError::SerialisationError);
 
     #[cfg(feature = "mock_serialise")]
-    let result = Ok(*data);
+    let result = Ok((*data).clone());
```

Comment:
why now need extra clone?

Reply:
It is an Rc, cannot move it out. Before it was a Box.

Reply:
I was confused after Qi's comment for a while as well - about why we need the deref with the clone, too - but we're returning a copy of the `Message` from within the `Rc` here, not the `Rc` itself. Makes sense.